src/main/scala/uk/gov/nationalarchives/consignmentexport/Validator.scala

```diff
@@ -0,0 +1,53 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.time.LocalDateTime
+import java.util.UUID
+
+import cats.effect.IO
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment.Files
+import cats.implicits._
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment.Files.Metadata
+import uk.gov.nationalarchives.consignmentexport.Validator.ValidatedFileMetadata
+
+class Validator(consignmentId: UUID) {
+  def validateConsignmentHasFiles(consignmentData: GetConsignment): IO[Files] = {
+    IO.fromOption(consignmentData.files.headOption)(new Exception(s"Consignment API returned no files for consignment $consignmentId"))
```

Comment:
What's the reason for wrapping this in an IO? There's no IO going on here, so I'd have expected a `Try` or `Either`.

Same goes for the other validation methods below.

Reply:
So it's to do with the for comprehension in the `Main` class. You can't have an `Either` and an `IO` in the same for comprehension. So this isn't valid
```
scala> for {
     | a <- IO("a")
     | b <- Right("b")
     | } yield b
       b <- Right("b")
         ^
On line 3: error: type mismatch;
        found   : scala.util.Either[Nothing,String]
        required: cats.effect.IO[?]

```
Now there are a whole load of type classes in the cats library which may be useful, I don't know but the fact that `IO.pure` exists for already evaluated "non IO" values makes me think that it's not always necessary for `IO` things to actually be IO.

You're right about the validation methods making more sense if they return an `Either` so I've changed those to return Either and then converted it into IO back in the `Main` class so the for comprehension is still valid. What do you think?

Reply:
That looks better to me, thanks! It makes it more obvious which parts of the code are actually doing IO, and which parts are doing a more straightforward data transformation.